server/src/internalClusterTest/java/org/opensearch/cluster/coordination/AwarenessAttributeDecommissionIT.java

```diff
@@ -56,6 +72,425 @@ public void cleanup() throws Exception {
         assertNoTimeout(client().admin().cluster().prepareHealth().get());
     }
 
```

Comment:
How can we verify that nodes are drained before initiating the decommissioning and once prepared for decommission no new request is being sent to these nodes?

Reply:
We can never certainly say if the node was drained completely or not. Although before initiating the decommission, we ensure weights for decommissioned zone is 0 but the metrics where we track HTTP requests doesn't get to zero. By the time we come up with an effective proposal for draining, we have implemented a 2 min wait after setting weights to 0 and executing decommission. 
Implemented as part of this PR - https://github.com/opensearch-project/OpenSearch/pull/4586